servicetalk-concurrent-api-internal/src/test/java/io/servicetalk/concurrent/api/publisher/AbstractPublishAndSubscribeOnTest.java

```diff
@@ -92,8 +96,56 @@ public void onComplete() {
                 });
         allDone.await();
 
-        verifyCapturedThreads(capturedThreads);
+        return verifyCapturedThreads(capturedThreads);
+    }
+
+    private static void recordThread(AtomicReferenceArray<Thread> threads, final int index) {
+        Thread was = threads.getAndUpdate(index, AbstractPublishAndSubscribeOnTest::updateThread);
+        assertThat("Thread already recorded at index: " + index, was, nullValue());
+    }
+
+    private static Thread updateThread(Thread current) {
+        assertThat(current, nullValue());
+        return currentThread();
+    }
+
+    public static AtomicReferenceArray<Thread> verifyCapturedThreads(AtomicReferenceArray<Thread> capturedThreads) {
+        for (int i = 0; i < capturedThreads.length(); i++) {
+            final Thread capturedThread = capturedThreads.get(i);
+            assertThat("No captured thread at index: " + i, capturedThread, notNullValue());
+        }
 
         return capturedThreads;
     }
+
+    public static TypeSafeMatcher<Thread> matchPrefix(String prefix) {
```

Comment:
Can we share `recordThread`, `verifyCapturedThreads`, `matchPrefix`, and `getNamePrefix` utilities? It's repeated 3 times now.

Reply:
I cleaned this up a bit and moved the `Matcher` to the new test utilities class. It seems more generally useful as there are similar usages all over the tests.